packages/dds/tree/src/simple-tree/core/treeNodeSchema.ts

```diff
@@ -41,16 +41,30 @@ export type TreeNodeSchema<
  * Non-class based schema can have issues with recursive types due to https://github.com/microsoft/TypeScript/issues/55832.
  * @system @sealed @public
  */
-export interface TreeNodeSchemaNonClass<
-	out Name extends string = string,
-	out Kind extends NodeKind = NodeKind,
-	out TNode extends TreeNode | TreeLeafValue = TreeNode | TreeLeafValue,
-	in TInsertable = never,
-	out ImplicitlyConstructable extends boolean = boolean,
-	out Info = unknown,
-> extends TreeNodeSchemaCore<Name, Kind, ImplicitlyConstructable, Info> {
-	create(data: TInsertable): TNode;
-}
+export type TreeNodeSchemaNonClass<
+	Name extends string = string,
+	Kind extends NodeKind = NodeKind,
+	TNode extends TreeNode | TreeLeafValue = TreeNode | TreeLeafValue,
+	TInsertable = never,
+	ImplicitlyConstructable extends boolean = boolean,
+	Info = unknown,
+	TConstructorExtra = never,
+> = TreeNodeSchemaCore<Name, Kind, ImplicitlyConstructable, Info, TInsertable> &
+	(undefined extends TConstructorExtra
```

Comment:
In this approach, TConstructorExtra can in theory be any type (right?) - though I don't think we have tests or a real use case that has TConstructorExtra be anything other than `undefined`. Would it be more appropriate to just have something like `TConstructorDefaulting extends boolean = false` then, and users set it to `true` to enable the optionality? Or do you anticipate us using types other than `undefined` here?

Reply:
I considered that. This approach seems more generally useful, and avoids annoyances like someone passing "Boolean" where we want true or false, and true being a logical subtype of false variance wise which is strange when used with extends clauses constraining schema (ex: if the default parameter is false, then schema with it set to true won't extend TreeNodeScehma, so the default would have to be boolean, or maybe TS would work out the odd variance from context, bit I don't trust that)

If a user overrides a constructor and adds extra options, it will work better now because of this (we still don't support that). Also there are use cases like recursive maps would allow you to construct them from object this way, if that didn't also break them like it does if we allow that for implicit construction.

Thus I think this is actually simpler typing wise than using a boolean toggle, and more general. THis does may me think I might have messed up the variance handling despite that since its contra variant, I'll go test it

Reply:
I added a variance test. Seems to work fine. I suspect both approaches would work ok, but personally I found the one I took simpler and more flexible.

Reply:
Cool - if it's just as simple/simpler but also more general, that seems like a win.